### ISSUE TITLE
Support MusicBrainz tags

### ISSUE THREAD

**Issue by fxthomas** [Libresonic/libresonic#163]
[MusicBrainz](http://musicbrainz.org/) is a great open option for auto-tagging music files in a library. Several tagging software make use of their services and write tags to the music files containing unique IDs that precisely identify albums and tracks. Among others, their own [Picard](https://picard.musicbrainz.org/) as well as [Beets](http://beets.io/) work very well.

If it exists, can we add a link to the related MusicBrainz entry on each album (next to the search links)?

**Comment by muff1nman** [comment]
This is a great idea, and at some point I'd like to take this further and use MusicBrainz, or similar for the unique key for music within libresonic and move away from using the file system path. The idea is that libresonic can be robust for folks who want to move their music around. Specifically I would like to use MusicBrainz to organize my folders/files.

**Comment by biconou** [comment]
Why not start by adding the MusicBrainz key beside the Libresonic key ?

**Comment by EugeneKay** [comment]
My feelings on organization clash a bit here - I feel very strongly that the filesystem path should not be ignored in favor of a third-party lookup service for metadata. This was a major factor behind my original decision to use Subsonic many moons ago - I knew that it wasn't going to mangle my collection(like iTunes).

Adding third-party metadata is nevertheless great for user experience reasons and filling incomplete data. I think this is a good feature to add, but we need to make sure not to screw up the basics ;-). I'll set this aside for the Future.

**Comment by fxthomas** [comment]
I agree with @EugeneKay : having brought my own collection back from an untagged mess over the last year and a half, it would be pretty painful to use metadata only in Librasonic unless they're squeaky clean for every file, which is not the case for everyone. Looking at the filesystem path first makes a lot of sense in this case.

However, just storing an ID in the database and adding links on properly tagged albums will go a long way for those of us who have proper tags, and should not cause any trouble at all.

In the future, this would open the way for some interesting features : show albums that have missing tracks, automatically download cover art from MB, show other albums from the same artist, show composer/performer information, etc.

Also, it seems that @muff1nman and I would both definitely benefit from something that can consolidate entries that have the same MusicBrainz ID, either in the media scanner directly or as a separate menu!

**Comment by muff1nman** [comment]
Just to clarify, I really like how low touch libresonic is on the physical files. In fact, I run mine over a readonly network filesystem. The thing is that my music collection is an utter mess, mostly because it came from iTunes and other hodge-podge sources. So I would like to organize it with an external tool, but this wouldn't work with how we identify music files right now. So I've been trying to think of a way to keep the existing behavior but also allow for libresonic to be a bit smarter when a file is moved within the same music folder path, but just under a different subfolder.
